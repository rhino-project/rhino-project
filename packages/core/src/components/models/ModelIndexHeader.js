import React from 'react';
import PropTypes from 'prop-types';

import { ModelFilters } from './ModelFilters';
import { ModelPager } from './ModelPager';
import { ModelSearch } from './ModelSearch';
import { useGlobalComponentForModel, useOverrides } from '../../hooks/overrides';
import { useModelIndexContext } from '../../hooks/controllers';

const defaultComponents = {
  ModelSearch,
  ModelFilters,
  ModelPager
};

export const ModelIndexHeaderBase = ({ overrides, ...props }) => {
  const { ModelSearch, ModelFilters, ModelPager } = useOverrides(
    defaultComponents,
    overrides
  );

  const { model } = useModelIndexContext();

  return (
    <div className={`model-index-header model-index-header-${model.model}`}>
      <div className="d-flex flex-column">
        {model.searchable === true && <ModelSearch {...props} />}
        <ModelFilters {...props} />
        <div className="d-flex flex-row">
          <div className="ms-auto">
            <ModelPager {...props} />
          </div>
        </div>
      </div>
    </div>
  );
};

ModelIndexHeaderBase.propTypes = {
  overrides: PropTypes.object
};

export const ModelIndexHeader = (props) =>
  useGlobalComponentForModel('ModelIndexHeader', ModelIndexHeaderBase, props);
