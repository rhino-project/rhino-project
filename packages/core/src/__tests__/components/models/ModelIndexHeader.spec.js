import { sharedModelTests } from './sharedModelTests';
import ModelIndexHeader from '../../../components/models/ModelIndexHeader';

describe('ModelIndexHeader', () => {
  sharedModelTests(ModelIndexHeader);
});
