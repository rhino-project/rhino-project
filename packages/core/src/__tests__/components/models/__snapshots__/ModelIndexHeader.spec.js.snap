// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelIndexHeader should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelIndexHeader should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
